runtime/src/snapshot_utils.rs

```diff
@@ -1538,30 +1535,43 @@ pub fn snapshot_bank(
     bank_snapshots_dir: impl AsRef<Path>,
     snapshot_archives_dir: impl AsRef<Path>,
     snapshot_version: SnapshotVersion,
-    archive_format: &ArchiveFormat,
+    archive_format: ArchiveFormat,
     hash_for_testing: Option<Hash>,
+    snapshot_type: Option<SnapshotType>,
 ) -> Result<()> {
-    let storages = root_bank.get_snapshot_storages(None);
+    let snapshot_storages = snapshot_type.map_or_else(SnapshotStorages::default, |snapshot_type| {
+        let incremental_snapshot_base_slot = match snapshot_type {
+            SnapshotType::IncrementalSnapshot(incremental_snapshot_base_slot) => {
+                Some(incremental_snapshot_base_slot)
+            }
+            _ => None,
+        };
+        root_bank.get_snapshot_storages(incremental_snapshot_base_slot)
+    });
     let mut add_snapshot_time = Measure::start("add-snapshot-ms");
-    add_bank_snapshot(&bank_snapshots_dir, root_bank, &storages, snapshot_version)?;
+    let bank_snapshot_info = add_bank_snapshot(
+        &bank_snapshots_dir,
+        root_bank,
+        &snapshot_storages,
+        snapshot_version,
+    )
+    .expect("failed to serialize bank snapshot to disk");
     add_snapshot_time.stop();
     inc_new_counter_info!("add-snapshot-ms", add_snapshot_time.as_ms() as usize);
 
-    // Package the relevant snapshots
-    let highest_bank_snapshot_info = get_highest_bank_snapshot_info(&bank_snapshots_dir)
-        .expect("no snapshots found in config bank_snapshots_dir");
-
-    let accounts_package = AccountsPackage::new_for_full_snapshot(
+    let accounts_package = AccountsPackage::new(
         root_bank,
-        &highest_bank_snapshot_info,
+        &bank_snapshot_info,
         bank_snapshots_dir,
         status_cache_slot_deltas,
         snapshot_archives_dir,
-        storages,
-        *archive_format,
+        snapshot_storages,
+        archive_format,
         snapshot_version,
         hash_for_testing,
-    )?;
+        snapshot_type,
+    )
+    .expect("failed to hard link bank snapshot into a tmpdir");
```

Comment:
Hmmm circling back to this `expect`, maybe we should return an error here and have ABS handle it so we're sure it exits properly? This ensures that ABS can set the exit variable so the validator shuts down when ABS shuts down.

Reply:
Handled offline.